src/Signal.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Signal
+ * Handles the various ways in which an action can be resolved. Currently,
+ * it handles values, promises, and error-first callbacks
+ */
+
+function Signal (action, params, next) {
+  let strategy = action.length >= 2 ? withCallback : withValue
```

Comment:
`function.length` returns the number of parameters a function has. This means:

```javascript
let single = (a) => {}
console.log(single.length) // => 1

let double = (a, b) => {}
console.log(double.length) // => 2
```

Sort of like with Mocha, I made it so that if two arguments are provided, it assumes callback notation